chore(team-org-roles): drop team org_role field in django

cathteng opened the pull request:
The `org_role` column on the `Team` model is no longer being used. All code relating to it has been removed. This is part of an ongoing effort to remove the org roles for teams feature https://github.com/getsentry/team-core-product-foundations/issues/51

Part 1 of removing columns according to https://develop.sentry.dev/database-migrations/#deleting-columns

Only removes the column from the state (Django).

Commit:
drop team org_role field in django

schew2381 reviewed (commented):
lgtm from product side, will wait for migration owner to approve

markstory reviewed (approved):
Looks good.

Commit:
remove assert

Commit:
remove prefetch

azaslavsky commented:
This is the first time since this feature was introduced that we’ve run across someone deleting a field. I’m working on a solution, and should have something up for review in a few hours.

azaslavsky committed:
feat(backup): Enable shimming deleted fields

We need to support deleted fields on Django model schemas for a number
of self-hosted released (currently 2). This means that even if we delete
a field on HEAD, we still need to make the import script aware that the
field could exit on imported JSON, and pre-process that JSON in such a
way that the field is removed in cases where the JSON was generated by
an old version that still contained the field.

There is no purpose in testing this code standalone. It passes tests in
 #65923, which is the first actual use of this shimming mechanism.

azaslavsky committed:
chore(team-org-roles): Add relocation shim for `org_role` field

azaslavsky commented:
I guess ~50 hours is technically still "a few". I've pushed two commits to this stack. The first is a general-case fix for schema field deletions. The second is the specific fix necessary for the field deletion done in this PR.

The first commit is a cherry-pick of a standalone PR at https://github.com/getsentry/sentry/pull/66248. Please wait for that to go in before merging this PR!

I've also made an issue, https://github.com/getsentry/sentry/issues/66247, for removing the shim, which will need to be completed once 24.4.0 is released in mid-April.

azaslavsky committed:
Mypy, don't be sad

cathteng commented on src/sentry/backup/imports.py:
```diff
@@ -150,14 +150,30 @@ def _import(
+    # the shim, noting in the comment which self-hosted release will trigger the removal.
+    deleted_fields: dict[str, set[str]] = {
+        # TODO(getsentry/sentry#66247): Remove once self-hosted 24.4.0 is released.
+        "sentry.team": {"org_role"}
```
ah i incorrectly rebased, thanks for catching!

Commit:
fix incorrect rebase

Commit:
add back deleted field
